### DISCUSSION TITLE
Sort directories first

### DISCUSSION THREAD

**Discussion by kputnam** [nvim-tree/nvim-tree.lua#1344] [Ideas]
In addition to the `sort_by` configuration, I'd like to be able to show directories before other types of files. So if `sort_by` is `name`, then I would see directories sorted by name, followed by other files sorted by name. This is currently possible with [NerdTree](https://github.com/preservim/nerdtree/blob/master/doc/NERDTree.txt#L1105), but I could not find a way to do this in NvimTree.

This type of feature will probably eventually face some scope creep, eg I want to sort certain things first or last. I think NERD Tree addresses it with its `NERDTreeSortOrder` configuration variable. There may be other ways to go about it though.

**Comment by alex-courtis** [comment]
I don't quite understand: directories are already (always) shown before files. See (old) screenshot:

https://raw.githubusercontent.com/kyazdani42/nvim-tree.lua/master/.github/screenshot2.png

autocommands.vim is shown after plugins.

**Reply by Shougo** [reply]
I think it is after the sort is customed.

**Reply by kputnam** [reply]
My mistake, you're correct @alex-courtis that directories are sorted first. I was simply disoriented  -- opening a file caused nvim-tree to automatically expand some directories, which made it look at first glance that things were mixed up.
